lib/routes.py

```diff
@@ -50,16 +50,43 @@ def get_groups():
 @socketio.on('join_group')
 def group_join(data):
   if find_group(data['access_code']):
-    room = data['access_code']
+    group = data['access_code']
+    room = None
     try:
-      Connection(group=room, sid=request.sid, created=datetime.utcnow).save()
-      join_room(room)
+      Connection(group=group, sid=request.sid, created=datetime.utcnow).save()
+      join_room(group)
+      room = matchmake(group)
     except:
       print('something went wrong with adding the connection')
-    send('Successfully Connected to Group', room=room)
+    if room:
+      send(f'Connected to {room}', room=room)
+    else:
+      send('Successfully Connected to Group', room=group)
   else:
     raise ConnectionRefusedError('Group Not Found')
 
+def matchmake(group):
+  match = None
+  if len(Connection.objects(group=group, waiting=True)) > 1:
+    for conn in Connection.objects(group=group, waiting=True):
+      if conn.sid != request.sid:
+        match = conn.sid
+        break
+      
```

Comment:
The below code is very easy to read.  Even with it's length I think it has an okay trade off.  I'm sure there is a way to refactor it smaller but that would probably lose a lot of it's readability.